persistent-mysql/Database/Persist/MySQL.hs

```diff
@@ -1055,24 +1056,51 @@ data SomeField record where
 -- the value that is provided. You can use this to increment a counter value.
 -- These updates only occur if the original record is present in the database.
 insertManyOnDuplicateKeyUpdate
-  :: ( PersistEntityBackend record ~ SqlBackend
-     , PersistEntity record
-     , MonadIO m
-     )
+  :: forall record backend m. 
+  ( backend ~ PersistEntityBackend record
+  , BackendCompatible SqlBackend backend
+  , PersistEntity record
+  , MonadIO m
+  )
   => [record] -- ^ A list of the records you want to insert, or update
   -> [SomeField record] -- ^ A list of the fields you want to copy over.
   -> [Update record] -- ^ A list of the updates to apply that aren't dependent on the record being inserted.
-  -> SqlPersistT m ()
+  -> ReaderT backend m ()
 insertManyOnDuplicateKeyUpdate [] _ _ = return ()
-insertManyOnDuplicateKeyUpdate records [] [] = insertMany_ records
+insertManyOnDuplicateKeyUpdate records [] [] = 
```

Comment:
I'm including this change in the PR because I haven't figured out a way to make `insertMany_` work in a non-breaking way. It works in projects where I can give a concrete type to the backend, but I would need to provide extra hints to type inference in the library now.